AER-4020 Added configuration option to set max number of workers available (#118)
aerius/taskmanager: README.md

Changing the `queueType` configuration parameter if the queues are already created won't work.
If `queueType` needs to be changed, the queues need to be deleted first.

The parameter `maxWorkersAvailable` (optional) can be set to the maximum number of workers the system can scale to when using automatically scaling of workers.
This is useful because when worker resources are automatically scaled it can cause issues in combination with a low `maxCapacityUse` parameter.
For example if worker scaling is based on the percentage workers being used it can mean the system will never scale up because the `maxCapacityUse` for certain input queues will never exceed the scaling threshold percentage.
Especially when the system runs with a low number of workers the `maxCapacityUse` is easily reached before the scaling threshold is reached.
By setting `maxWorkersAvailable` the scheduler will determine if tasks already have reached the `maxCapacityUse` value based on the `maxWorkersAvailable` value and not the actual number of workers running.

> It will than schedule tasks beyond the `maxCapacityUse` assuming this would trigger the system to scale up if needed,

Reviewer: ```suggestion
It will then schedule tasks beyond the `maxCapacityUse`, assuming this will trigger the system to scale up as needed,
```

> and only for a short amount of time claim more resources for a specific queue than would be allowed by the `maxCapacityUse` percentage.

Reviewer: only for a short time if the `maxWorkersAvailable` matches the actual system max scaling. If that isn't configured correctly, it might end up claiming more for longer periods (bit nitpicky perhaps).

Author: I've rewritten it to also include about this.
